pkg/services/tree/service.go

```diff
@@ -64,6 +64,18 @@ func (s *Service) Add(_ context.Context, req *AddRequest) (*AddResponse, error)
 		return nil, err
 	}
 
+	var resp *AddResponse
+	var outErr error
+	err = s.forEachNode(ctx, cid, func(c TreeServiceClient) bool {
+		resp, outErr = c.Add(ctx, req)
```

Comment:
why dont we try to check err here and continue retransmitting to another node?

Reply:
Because the error can happen and if we are here, the connection was already established, so the error is most likely should be returned.